### PR TITLE
Move stats code into WPiOS as a dynamic framework

### PR DESCRIPTION
**Pull request by astralbodies**
Issue #7120 

This PR does a bunch of things and shouldn't require a ton of testing but should be reviewed carefully for an understanding of what's happened to the project.

* A new project was created called WordPressComStatsiOS. This project has two targets - a dynamic framework and a test target. All of the code and configuration settings were made to mirror what the original stats project podspec did. I chose to not make a resource bundle like I did in the podspec so a few things had to change in WPiOS to load the SiteStats Storyboard properly.
* Stats requires CocoaPods because of things like networking, logging, etc. The Podfile was updated to support this new project and shared pod definitions were placed into a macro of sorts to make version bumps easier. The stats project is in it's own root-level target which also means the dependency of the project on the main WordPress target is unknown to CocoaPods.
* Manually configured the main target to embed the framework. Also added the framework to the linker for the today widget.
* Added the stats unit tests to the scheme for unit tests in the main target.

## To Test
1. Update your local CocoaPods repo with `pod repo update`.
2. Optionally clean your `DerivedData` folder.
3. Build & test main WordPress target. All tests should pass - UI tests are turned off now with this PR.
4. Run main WordPress target.
5. View stats for a site. Make sure nothing crashes, icons load on screens.
6. Try viewing stats via a stats notification. Make sure nothing crashes.
7. Try viewing stats on a published post by tapping the stats button in a post list.
8. Stop the running app.
9. Switch to the WordPressComStatsiOS scheme and run unit tests.

See if you can break it any other way.

Needs review: @elibud, @nheagy, @koke, @frosty, @kurzee

### CODE REVIEW

**Comment by astralbodies** [comment]
I will fix the code signing issues with Buddybuild and get the build to a green state.

**Comment by astralbodies** [comment]
I fixed the issues with the Internal and Alpha builds - I had to copy the build configuration names (Release-Internal and Release-Alpha) to get the stats framework to build properly. I'm sure there is some way I could have mapped configuration names between each ... not going to worry about it now. 😄 

I also just moved the `WordPressComStatsiOS` project up one folder so it's on par with the visualization of the structure in the Workspace.

**Comment by elibud** [comment]
This is great @astralbodies, functionality wise everything seems perfect.

For point 6, as far as I can see in the code, only a New Follower push notification has the potential to trigger stats, is that right? However it doesn't look like with the current configuration of notes that code path is reachable. @jleandroperez do you remember seeing a Stats screen from a New Follower notification?

Running the tests was failing for me at some point with `Module OCMock not found` but started working after I changed the device target and then went back to the Simulator (no idea why).

File: WordPress/WordPress.xcodeproj/xcshareddata/xcschemes/WordPress Internal.xcscheme [hunk 1]
```diff
@@ -81,11 +86,17 @@
...
+         </CommandLineArgument>
+      </CommandLineArguments>
       <EnvironmentVariables>
          <EnvironmentVariable
-            key = "NSZombieEnabled"
-            value = "YES"
-            isEnabled = "YES">
+            key = "OS_ACTIVITY_MODE"
+            value = "disable"
+            isEnabled = "NO">
```
Comment (frosty) [inline WordPress/WordPress.xcodeproj/xcshareddata/xcschemes/WordPress Internal.xcscheme]: Didn't we remove this setting?

Comment (astralbodies) [inline WordPress/WordPress.xcodeproj/xcshareddata/xcschemes/WordPress Internal.xcscheme]: I thought we did too...

**Comment by astralbodies** [comment]
I did fix the issue that @frosty brought up regarding unit tests not being runnable. Turns out I didn't fix the one file reference after I moved the stats project up one directory level.

File: WordPress/WordPress.xcodeproj/xcshareddata/xcschemes/WordPress Internal.xcscheme (hunk 1 above)
Comment (astralbodies) [inline WordPress/WordPress.xcodeproj/xcshareddata/xcschemes/WordPress Internal.xcscheme]: Oh actually I left the option there but disabled it so if we need to turn it on it's there for us to turn on/off.

**Comment by koke** [comment]
> A new project was created called WordPressComStatsiOS

I'm curious, why not add the targets directly to the WordPress project?

**Comment by koke** [comment]
I think running tests on the WordPress target should also run the stats tests, and it seems like the scheme is set up for that but it says they're missing, and they're not running:

screen shot 2017-05-10 at 09 37 22

**Comment by frosty** [comment]
@koke Did you pull the latest update from @astralbodies? I was seeing the same issue as you, but  it's fixed for me now. `WordPressComStatsiOSTests` are available, and run along with the main target tests. Also try:

* Deleting the workspace, then `pod install`.
* Deleting `WordPress/WordPress.xcodeproj/xcuserdata` and doing the dance again

**Comment by astralbodies** [comment]
@koke 
> I'm curious, why not add the targets directly to the WordPress project?

This was the least impacting change (in my mind) to bring the project in as a separate project to simulate how CocoaPods does things separately. I didn't want any of the code to be visible to the main WordPress targets to prevent the tempting maneuver of adding a needed class into a second target instead of importing the framework.

Easy enough to change going forward as I suspect we will pull stats into the main app and get rid of the framework altogether.

**Comment by astralbodies** [comment]
Merging in develop, fixed one conflict, and will merge when Buddybuild gives me green. Thanks everyone for taking the time to review!

**Pull request merged**